src/geocat/viz/contourf/contourf.py

```diff
@@ -1,81 +1,435 @@
+                                                  self.ylim[0], self.ylim[1]
+                                              ],
+                                              extend=self.cbextend)
+
+    def _add_contour_labels(self,
+                            ax,
+                            lines,
+                            contourlabels=None,
+                            background=True,
+                            fontsize=12):
+        """Add contour line labels with an optional white background to the
+        figure.
+
+        Kwargs:
+
+            contourbackground (:obj:`bool`): Whether a white background for the contour labels will be drawn. Default False.
+
+            contourfontsize (:obj:`int`): Font size of the contour line labels. Default 12.
+
+            contourlabels (:obj:`list` or :class:`numpy.ndarray`): List or array of labels to use for contour line labels.
+
+            drawcontourlabels(:obj:`bool`): Whether add contour line labels to the figure.
+
+            flevels (:obj:`list` or :class:`numpy.ndarray`): List or array of levels to be passed into matplotlib's contourf function.
+
+            manualcontourlabels (:obj:`bool`): Whether contour line labels should be manually drawn. Default False.
+        """
+
+        # Update argument definitions
+        if self.contourfontsize is not None:
+            fontsize = self.contourfontsize
+
+        # If level range is less than 1, set labels with smaller differences between them
+        if (self.levels[-1] - self.levels[0]) <= 1:
+            fmt = '%0.2f'
+        else:
+            fmt = '%d'
+
+        # Set contour line labels based on inputted arguments. Depending on which arguments are included or not, contour labels must be created differently.
+        if self.contourlabels is None:
+            ax.clabel(lines,
+                      fontsize=fontsize,
+                      fmt=fmt,
+                      inline=True,
+                      colors="black")
+        elif self.manualcontourlabels is False:
+            ax.clabel(lines,
+                      contourlabels,
+                      fontsize=fontsize,
+                      fmt=fmt,
+                      inline=True,
+                      colors="black")
+        elif self.manualcontourlabels is True:
+            ax.clabel(lines,
+                      fontsize=fontsize,
+                      fmt=fmt,
+                      inline=True,
+                      manual=contourlabels,
+                      colors="black")
+        else:
+            raise AttributeError(
+                "Manualcontourlabels, if set, must be True or False.")
+
+        # Add white background to contour line labels
+        if background is True:
+            [
+                txt.set_bbox(dict(facecolor='white', edgecolor='none', pad=2))
+                for txt in lines.labelTexts
+            ]
 
     def _estimate_flevels(self):
         # TODO: flesh out
```

Comment:
Do we have some of this already from Jiaqi's work? Maybe these are something that could go in a plot utils folder, as it may be more broadly useful than just contours

Reply:
I believe Jiaqi has a separate pull request with this! We can definitely add it to NCL Plot if we want it in that class or in a separate class within plot utils.